cpp/src/arrow/array/data.cc

```diff
@@ -214,6 +214,14 @@ void ArraySpan::SetMembers(const ArrayData& data) {
     this->buffers[i] = {};
   }
 
+  if (type_id == Type::STRING_VIEW || type_id == Type::BINARY_VIEW) {
+    // store the span of character buffers in the third buffer
+    this->buffers[2].data =
+        const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(data.buffers.data() + 2));
+    this->buffers[2].size =
+        static_cast<int64_t>(data.buffers.size() - 2) * sizeof(std::shared_ptr<Buffer>);
```

Comment:
It's used! We can't access index/offset views without all the character buffers. Since ArraySpan has a compile-time limit of three buffers, there's nowhere else to put these